tests/administrators/test_data.py

```diff
@@ -0,0 +1,171 @@
+import asyncio
+from asyncio import gather
+
+import pytest
+
+from syrupy.filters import props
+from virtool_core.models.roles import AdministratorRole
+
+from virtool.administrators.oas import UpdateUserRequest, UpdateAdministratorRoleRequest
+from virtool.authorization.relationships import AdministratorRoleAssignment
+from virtool.data.errors import ResourceNotFoundError
+from virtool.users.db import validate_credentials
+
+
+@pytest.mark.parametrize("term", [None, "test_user", "missing-handle"])
+@pytest.mark.parametrize("administrator", [True, False, None])
+async def test_find_users(
+    spawn_client,
+    authorization_client,
+    fake2,
+    snapshot,
+    data_layer,
+    administrator,
+    term,
+):
+
+    group_1 = await fake2.groups.create()
+    group_2 = await fake2.groups.create()
+
+    user_1, user_2, _ = await asyncio.gather(
+        fake2.users.create(
+            handle="test_user", groups=[group_1, group_2], primary_group=group_1
+        ),
+        fake2.users.create(),
+        fake2.users.create(),
+    )
+
+    await authorization_client.add(
+        AdministratorRoleAssignment(user_1.id, AdministratorRole.BASE),
+        AdministratorRoleAssignment(user_2.id, AdministratorRole.FULL),
+    )
+
+    assert (
+        await data_layer.administrators.find_users(
+            1, 25, term=term, administrator=administrator
+        )
+        == snapshot
+    )
+
+
+@pytest.mark.parametrize("administrator", [True, False])
+async def test_get_user(
+    spawn_client, authorization_client, fake2, snapshot, data_layer, administrator
+):
+    group = await fake2.groups.create()
+    user = await fake2.users.create(
+        handle="test_user", groups=[group], primary_group=group
```

Comment:
Why is it necessary to hard code `handle` here?

Reply:
Good catch, it isn't